src/components/Layout/Navigation.tsx

```diff
@@ -0,0 +1,161 @@
+import { ConnectionStatus, NavigationProps, NavigationLinkProps } from './types';
+import { NavLink } from 'react-router-dom';
+import {
+  ArrowUpOutlined,
+  CheckOutlined,
+  ClockCircleOutlined,
+  DoubleLeftOutlined,
+  DoubleRightOutlined,
+  LoginOutlined,
+  WarningOutlined,
+} from '@ant-design/icons';
+
+function Navigation({
+  navIsExpanded,
+  setNavIsExpanded,
+  bottomNavigation,
+  topNavigation,
+  gcStatus,
+}: NavigationProps) {
+  const linkClasses =
+    'block cursor-pointer overflow-hidden h-12 px-6 py-3 text-[15px] transition-none';
+  const linkClassesSelected = `${linkClasses} bg-crate-blue text-white hover:text-white`;
+  const linkClassesUnselected = `${linkClasses} text-crate-navigation-fg hover:bg-neutral-700/50 hover:text-crate-navigation-fg`;
+
+  const linkKey = (type: string, path: string): string => `${type}_${path}`;
+
+  const drawLinkInner = (
+    icon: JSX.Element,
+    label: JSX.Element,
+    navIsExpanded: boolean,
+    isExternalLink: boolean = false,
+  ): JSX.Element => {
+    return (
+      <div className="flex flex-nowrap gap-2 align-center max-w-[300px] mx-auto w-full whitespace-nowrap">
+        <div className="text-center text-sm h-6 w-6">{icon}</div>
+        {navIsExpanded && (
+          <div className="flex flex-nowrap gap-1 items-start overflow-hidden">
+            {label}
+            {isExternalLink && <ArrowUpOutlined className="opacity-50 rotate-45" />}
+          </div>
+        )}
+      </div>
+    );
+  };
+
+  const drawExpandCollapse = (): JSX.Element => (
+    <div
+      className={`${linkClassesUnselected} hidden md:block`}
+      onClick={() => setNavIsExpanded(!navIsExpanded)}
+    >
+      {drawLinkInner(
+        navIsExpanded ? <DoubleLeftOutlined /> : <DoubleRightOutlined />,
+        navIsExpanded ? <span>Collapse</span> : <span>Expand</span>,
+        navIsExpanded,
+      )}
+    </div>
+  );
+
+  const drawGCConnectionStatus = (status?: ConnectionStatus): JSX.Element => {
```

Comment:
Are we planning to have this also in `cloud-ui`? Otherwise I think that this can be handle by the `crate-gc-admin`. We can add an additional prop as a generic `ReactNode` and simply render it, to avoid the `enum ConnectionStatus` duplication.

Reply:
I'd think the GC connection status is something mostly for debugging / testing and not something we would want to show to the user. Especially in the cloud we probably want to hide the existence completely

Reply:
> I'd think the GC connection status is something mostly for debugging / testing and not something we would want to show to the user. Especially in the cloud we probably want to hide the existence completely

The GC status thing is an optional parameter: it wouldn't be used in the cloud UI.

Reply:
> Are we planning to have this also in `cloud-ui`? 
Yes we are (though it might not get used immediately). The reason why gcStatus is passed in (or not) is so the expand / collapse feature works.